### DISCUSSION TITLE
Why a model with relation to another model via non-null foreign key has nullable type?

### DISCUSSION THREAD

**Discussion by arslancharyev31** [prisma/prisma#9137] [Q&A] [answered]
Take the following schema adapted from the starter project:
```prisma
model Post {
  id        Int     @id @default(autoincrement())
  content   String
  author    User   @relation(fields: [authorId], references: [id])
  authorId  Int
}

model User {
  id    Int     @id @default(autoincrement())
  name  String
  posts Post[]
}
```

This results in a TS client that has following return types:
```typescript
let author = await prisma.post.findUnique({
  rejectOnNotFound: true,
  where: {id: 1}
})        // type: Post
.author() // type: User | null
```

We can see that the return value of `findUnique` call will have non-null type `Post` since `rejectOnNotFound` was set to `true`.
But the subsequent call to `authors()` yields `User | null` type. This to me seems like an erroneous union with null since we've specified non-null foreign key to the `User` model.

This nullable return type means that in a type-safe environment in order to return an `author` object with non-null type one has to either use non-null assertion operators:
```typescript
author = author! // type: User
```
 Or perform manual runtime validation with exceptions:
```typescript
if(!author)
  throw Error("Impossible")
author // type: User
```
Both of which are not ideal, and are unnecessary.

I'm not sure if this is bug or something that I missed in my schema definition, hence I post this in Discussions.

**Comment by ryands17** [comment, marked as answer]
This looks like a bug. It would be great if you could create an issue [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) so that we can look into this. In the meantime, you can use `select` or `include` like this to fetch the relation and it should have the correct type:

```ts
let post = await prisma.post.findUnique({
  rejectOnNotFound: true,
  where: {id: 1},
  include: { author: true }
}) 

console.log(post.author)
```

**Reply by arslancharyev31** [reply]
Thanks for the swift answer. I've created the corresponding issue.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
